internal/deploys/models.go

```diff
@@ -87,3 +72,51 @@ func (db dataStore) listAll() ([]ServiceInstance, error) {
 
 	return serviceInstances, nil
 }
+
+// Deploy is the type  defining the database model for a deployment. It is an association
+// between a service instance and a build
+type Deploy struct {
+	ID                int
+	ServiceInstanceID int
+	ServiceInstance   ServiceInstance `gorm:"foreignKey:ServiceInstanceID;references:ID"`
+	BuildID           int
+	Build             builds.Build `gorm:"foreignKey:BuildID;references:ID"`
+	CreatedAt         time.Time
+	UpdatedAt         time.Time
+}
+
+type deployStore interface {
+	createDeploy(buildID, serviceInstanceID int) (Deploy, error)
+	getDeploysByServiceInstance(serviceInstanceID int) ([]Deploy, error)
+}
+
+func newDeployStore(dbConn *gorm.DB) dataStore {
+	return dataStore{dbConn}
+}
+
+func (db dataStore) createDeploy(buildID, serviceInstanceID int) (Deploy, error) {
+	newDeploy := Deploy{
+		ServiceInstanceID: serviceInstanceID,
+		BuildID:           buildID,
+	}
+
+	if err := db.Create(&newDeploy).Error; err != nil {
+		return Deploy{}, err
+	}
+
+	return newDeploy, nil
+}
+
+func (db dataStore) getDeploysByServiceInstance(serviceInstanceID int) ([]Deploy, error) {
+	var deploys []Deploy
+
+	err := db.Preload("ServiceInstance").
+		Preload("ServiceInstance.Service").
+		Preload("ServiceInstance.Environment").
+		Preload("Build").
+		Preload("Build.Service").
```

Comment:
I haven't really looked too much into the lazy-loading behavior of GORM but maybe here is a good time to do that, this just feels massive, also we're loading service twice. Or least (considering time) maybe put a TODO comment to note we reaaaaaally should make sure this is won't explode us.